src/packages.jl

```diff
@@ -408,6 +410,111 @@ function build(name::String; quiet::Bool = false,
+all of its `NeededOtherPackages`, recursively.
+All keyword arguments are passed on to [`build`](@ref).
+"""
+function build_recursive(name::String; quiet::Bool = false,
+                             debug::Bool = false,
+                             pkgdir::AbstractString = DEFAULT_PKGDIR[])
+  # point PackageManager to the given pkg dir
+  Globals.PKGMAN_CustomPackageDir = GapObj(pkgdir)
+  mkpath(pkgdir)
+
+  todo = Set{String}((name,))
+  done = Set{String}()
+  while !isempty(todo)
+    dep = pop!(todo)
+    dep in done && continue
+
+    res = build(dep; quiet, debug, pkgdir)
+    res || return false
+    push!(done, dep)
+    gdep = GapObj(dep)
+    allinfo = collect(Globals.PackageInfo(gdep))
+    installpath = Globals.TestPackageAvailability(gdep)
+    if installpath != Globals.fail
+      installpath === true && continue
+      info = only(filter(info -> info.InstallationPath == installpath, allinfo))
+    else
+      info = first(allinfo) # not sure what to do here if there are multiple versions available
+    end
+    for (needed, _) in info.Dependencies.NeededOtherPackages
+      push!(todo, String(needed))
+    end
+  end
+  return true
+end
+
+"""
+    test(name::String)
+
+Test the GAP package with name `name`.
+
+The function uses [the function `TestPackage`](GAP_ref(ref:TestPackage)).
+
+Use-sites of this function should verify that the `Test` package is loaded
+before calling this function, as this function might be moved
+to a package extension that depends on the `Test` package in the future.
+"""
+function test(name::String)
+  global disable_error_handler
+  
+  function with_gap_var(f, name::String, val)
+    gname = GapObj(name)
+    old_value = Globals.ValueGlobal(gname)
+    Globals.MakeReadWriteGlobal(gname)
+    Globals.UnbindGlobal(gname)
+    Globals.BindGlobal(gname, val)
+    try
+        f()
+    finally
+      Globals.MakeReadWriteGlobal(gname);
+      Globals.UnbindGlobal(gname);
+      Globals.BindGlobal(gname, old_value);
+    end
+  end
+
+  error_occurred = false
+  ended_using_QuitGap = false
+  function fake_QuitGap(code)
+    if code != 0
+      error_occurred = true
+    end
+    ended_using_QuitGap = true
+  end
+
+  disable_error_handler[] = true
+  try
+    with_gap_var("ERROR_OUTPUT", Globals._JULIAINTERFACE_ORIGINAL_ERROR_OUTPUT) do
+      with_gap_var("QuitGap", fake_QuitGap) do
+        with_gap_var("QUIT_GAP", fake_QuitGap) do
+          with_gap_var("ForceQuitGap", identity) do
+            with_gap_var("FORCE_QUIT_GAP", identity) do
+              result = Globals.TestPackage(GapObj(name))
+              @test ended_using_QuitGap || result == true
+              # Due to the hack above, we run into an error in TestPackage that is usually unreachable.
+              # In the case of a `QuitGap` call, we thus don't check for `result == true`.
+            end
+          end
+        end
+      end
+    end
+  finally
+    disable_error_handler[] = false
+  end
+
+  @test !error_occurred
```

Comment:
Could we just do this and not use `@test` here and rely on the caller to do `@test GAP.Packages.test("bla")` ?
```suggestion
  return !error_occurred && (ended_using_QuitGap || result == true)
```